File: namastack-outbox-examples/namastack-outbox-example-testcontainers-stable/src/test/java/io/namastack/demo/stabletest/TestWithPostgres.java
package io.namastack.demo.stabletest;

import org.springframework.boot.testcontainers.service.connection.ServiceConnection;
import org.testcontainers.junit.jupiter.Container;
import org.testcontainers.postgresql.PostgreSQLContainer;

public interface TestWithPostgres {

  @Container
  @ServiceConnection
  @SuppressWarnings("resource")
  PostgreSQLContainer postgresContainer = new PostgreSQLContainer("postgres:18.3-alpine");
}

File: namastack-outbox-examples/namastack-outbox-example-testcontainers-flaky/src/test/kotlin/io/namastack/demo/flakytest/TestWithPostgres.kt
package io.namastack.demo.flakytest

import org.springframework.boot.testcontainers.service.connection.ServiceConnection
import org.testcontainers.junit.jupiter.Container
import org.testcontainers.postgresql.PostgreSQLContainer

interface TestWithPostgres {

    companion object {
        @Container
        @ServiceConnection
        @JvmField
        val postgresContainer: PostgreSQLContainer = PostgreSQLContainer("postgres:18.3-alpine")
    }
}

